Zend/zend_alloc.h

```diff
@@ -26,14 +26,16 @@
 #include "../TSRM/TSRM.h"
 #include "zend.h"
 
-#ifndef ZEND_MM_ALIGNMENT
-# define ZEND_MM_ALIGNMENT Z_UL(8)
-# define ZEND_MM_ALIGNMENT_LOG2 Z_L(3)
+#ifndef CONFIG_ZEND_MM_ALIGNMENT
+# define ZEND_MM_ALIGNMENT (size_t)8
+# define ZEND_MM_ALIGNMENT_LOG2 (size_t)3
 #elif ZEND_MM_ALIGNMENT < 4
```

Comment:
This check should be checking for CONFIG_ZEND_MM_ALIGNMENT.

Overall, I don't really like this change. configure should always determine an alignment (there is even a fallback for cross-compiling). I believe ZEND_MM_ALIGNMENT is currently only not defined on windows, where we should define it in config.w32.h.in.

I think we would be better off moving the check for alignment < 4 into https://github.com/php/php-src/blob/941b55fe6397cd7aa9e843c9209a782172518386/Zend/Zend.m4#L259 and add `u` suffixes to https://github.com/php/php-src/blob/941b55fe6397cd7aa9e843c9209a782172518386/Zend/Zend.m4#L269 and drop the code here entirely.

Reply:
Gah, it looks like just adding `u` will make ZEND_MM_ALIGNED_SIZE incorrect. It works right now because ZEND_MM_ALIGNMENT_MASK is a signed integer, so the mask is signed extended. It wouldn't work for an unsigned integer. So looks like that does need to be size_t specifically if we switch it to unsigned.

Reply:
ACK, I'll look into making the changes in config and dropping this.